js/data/buffer/circle_vertex_buffer.js

```diff
@@ -1,41 +1,31 @@
 'use strict';
 
 var util = require('../../util/util');
-var Buffer = require('./buffer');
-
-module.exports = CircleVertexBuffer;
-
-/**
- * This contains the data that displays circle markers on the map,
- * including their centerpoint
- */
-function CircleVertexBuffer(buffer) {
-    Buffer.call(this, buffer);
+var Buffer2 = require('../buffer2');
+
+function CircleVertexBuffer(options) {
+    Buffer2.call(this, options || {
+        type: Buffer2.BufferType.ELEMENT,
```

Comment:
This should be `Buffer2.BufferType.VERTEX`. 

There are two kinds of buffers: element buffers and vertex buffers. 

 - Vertex buffers encode a set of user-defined attributes per vertex.
 - Element buffers encode references to sets of three vertices that constitute a triangle. 

(Apologies if this is review for you!)